lib/api/redfish-1.0/task-service.js

```diff
@@ -0,0 +1,145 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var injector = require('../../../index.js').injector;
+var redfish = injector.get('Http.Api.Services.Redfish');
+var waterline = injector.get('Services.Waterline');
+var Promise = injector.get('Promise');
+var _ = injector.get('_');
+var moment = require('moment');
+
+module.exports = {
+    taskServiceRoot: taskServiceRoot,
+    listTasks: listTasks,
+    getTask: getTask,
+    getSystemTasks: getSystemTasks
+};
+
+var dataFactory = function(identifier, dataName) {
+    var statusMap = {
+        'valid': 'Running',
+        'succeeded': 'Completed',
+        'cancelled': 'Killed',
+        'failed':  'Exception'
+    };
+
+    switch(dataName)  {
+        case 'graphObjects':
+            return Promise.map(waterline.graphobjects.find({}), function(graph) {
+                    return { 
+                        state: statusMap[graph._status],
+                        id: graph.id,
+                        status: 'OK',
+                        startTime: moment.utc(graph.createdAt).format(),
+                        endTime: (graph._status !== 'valid') ? 
+                            moment.utc(graph.updatedAt).format() : 
+                            '',
+                        name: graph.name,
+                        systemId: graph.node
+                    };
+                });
+    }
+};
+
+function taskServiceRoot(req, res) {
+    var options = {};
+    options.basepath = req.swagger.operation.api.basePath;
+    options.templateScope = ['global'];
+    options.url = req.url;
+    options.dateTime = moment.utc().format();
+    options.systems = [];
+
+    Promise.all([ dataFactory('', 'graphObjects') ])
+        .spread(function(graphs) {
+            _.forEach(graphs, function(graph) {
+                if (graph.systemId && -1 === options.systems.indexOf(graph.systemId))  {
+                    options.systems.push(graph.systemId);
+                }
+            });
+            options.graphs = graphs;
+            return redfish.render('redfish.1.0.0.taskservice.1.0.0.json', 
+                 'TaskService.1.0.0.json#/definitions/TaskService',
```

Comment:
I think @WangWinson 's concern here is if we move it from database to filesystem, it will be a way to reduce the RackHD footprint, I know some people is working on this. 

Meanwhile, put it into filesytem will get the dynamic loading feature, but our current database design only load templates one time when app runs, if we put a new template file (especially during debugging), we have to reboot the app to let the new template file takes effect.

Of course, this discussion is not push you to change it in this PR, first we need a discussion, if agree, we can do it in another PR.